src/components/shared/ScrollToFieldError/index.js

```diff
@@ -0,0 +1,47 @@
+import { useEffect } from 'react';
+
+export const ScrollToFieldError = ({
+  scrollBehavior = { left: 0, behavior: 'smooth' },
+  useFormikContext,
+  fieldsOrder = [],
+}) => {
+  const { submitCount, isValid, errors } = useFormikContext();
+
+  useEffect(() => {
+    if (isValid) {
+      return;
+    }
+
+    const fieldErrorNames = Object.keys(errors);
+
+    const firstFieldErrorName = fieldsOrder.find((fieldName) =>
+      fieldErrorNames.includes(fieldName),
+    );
+    if (!firstFieldErrorName) {
+      return;
+    }
+
+    const element = document.querySelector(`input[name='${firstFieldErrorName}']`);
+    if (!element) {
+      return;
+    }
+
+    // unfocus the active element (in mobile, close keyboard)
+    if (document.activeElement) {
+      document.activeElement.blur();
+    }
+
+    // Scroll to first known error into view
+    setTimeout(() => {
+      window.scroll({
+        top: element.getBoundingClientRect().top + window.scrollY - 30,
+        ...scrollBehavior,
+      });
+    }, 150);
+    setTimeout(() => {
+      element.focus();
+    }, 500);
+  }, [submitCount]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Do you really need to disable exhaustive-deps validation? is not possible to add the right dependencies?

Reply:
Yes, I need it. Because otherwise it would give warnings in console.

Reply:
You will not get warnings if you add all the dependencies to the array, right?

Reply:
Yes, if I add all the dependencies I will not get warnings. But I just need that the useEffect it execute every time the user clicks the submit, for that I just add `submitCount` as dependency

Reply:
Just for the record:

We have talked about it, the `errors` array changes more often than the `submitCount` and we do not want to run this code each time that the `errors` array has a change.

In my opinion, we should search for a way to include the dependency, for example with a _ref_, in order to make this code less error-prone. But @jcamposmk alternative has the advantage of simplicity: just adding the comment `// eslint-disable-line react-hooks/exhaustive-deps` is enough.